clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -512,8 +513,15 @@ class CIRFuncLowering : public mlir::OpConversionPattern<mlir::cir::FuncOp> {
         resultType ? resultType : mlir::LLVM::LLVMVoidType::get(getContext()),
         signatureConversion.getConvertedTypes(),
         /*isVarArg=*/fnType.isVarArg());
-    auto fn = rewriter.create<mlir::LLVM::LLVMFuncOp>(op.getLoc(), op.getName(),
-                                                      llvmFnTy);
+    // LLVMFuncOp expects a single FileLine Location instead of a fused
+    // location.
+    auto Loc = op.getLoc();
+    if (Loc.isa<mlir::FusedLoc>()) {
+      auto FusedLoc = Loc.cast<mlir::FusedLoc>();
+      Loc = FusedLoc.getLocations()[0];
+    }
+    auto fn =
```

Comment:
There might be other locations types besides `mlir::FusedLoc`, please also check for `FileLine` explicitly and add a generic `else` that asserts.

Reply:
> There might be other locations types besides `mlir::FusedLoc`, please also check for `FileLine` explicitly and add a generic `else` that asserts.

A FileLine loc should also work with the current logic. Just added an assert. How does that look to you?

Reply:
Great!